libs/community/langchain_community/chat_models/symblai_nebula.py

```diff
@@ -0,0 +1,277 @@
+import json
+import requests
+from typing import Any, AsyncIterator, Dict, Iterator, List, Optional
+from aiohttp import ClientSession
+
+from langchain_core.callbacks import (
+    AsyncCallbackManagerForLLMRun,
+    CallbackManagerForLLMRun,
+)
+from langchain_core.language_models.chat_models import (
+    BaseChatModel,
+    agenerate_from_stream,
+    generate_from_stream,
+)
+from langchain_core.messages import (
+    AIMessage,
+    AIMessageChunk,
+    BaseMessage,
+    ChatMessage,
+    HumanMessage,
+    SystemMessage,
+)
+
+from langchain_core.outputs import ChatGeneration, ChatGenerationChunk, ChatResult
+from langchain_core.pydantic_v1 import Field, SecretStr, root_validator
+from langchain_core.utils import convert_to_secret_str, get_from_dict_or_env
+
+def _convert_role(role: str) -> str:
+    map = {"ai": "assistant", "human": "human", "chat": "human"}
+    if role in map:
+        return map[role]
+    else:
+        raise ValueError(f"Unknown role type: {role}")
+
+
+def _format_nebula_messages(messages: List[BaseMessage]) -> Dict[str, Any]:
+    system = ""
+    formatted_messages = []
+    for message in messages[:-1]:
+        if message.type == "system":
+            system = message.content
+        else: 
+            formatted_messages.append(
+                {
+                    "role": _convert_role(message.type),
+                    "text": message.content,
+                }
+            )
+            """
+            The first message in formatted_messages must be "human"
+            """
+            if formatted_messages[0]["role"] != "human":
+                print("The first message after system prompt must be a human message")
```

Comment:
there should be no `print` or `exit()` in the code. If needed raise an exception